### ISSUE TITLE
improve realloc

### ISSUE THREAD

**Issue by maddin200** [BioMedIA/MIRTK#474]
Modules\IO\src\nifti\nifti2_io.cc line 8274
maybe better is:
 newbuf = realloc(buf, ...
 if (newbuf != NULL)
 {

**Comment by schuhschuh** [comment]
Indeed, there a few more instances of this. Thing is just that all these rather minor bugs are in the source code of a third-party library that is just for convenience included in the MIRTK source tree to minimize the required pre-installed dependencies. It would be better if these bugs would be reported and fixed upstream by the developers of the NIfTI library. Otherwise, when we replace the included sources with a newer upstream version, we will have to make sure to replicate our fixes/modifications of these files.

https://nifti.nimh.nih.gov/pub/dist/src/README

**Comment by maddin200** [comment]
I will have a look.
